File: src/main/java/net/liyze/basin/core/BasinBoot.java
package net.liyze.basin.core;

/**
 * Basin's Plugin/Boot SPI
 */
public interface BasinBoot {
    /**
     * Runs on basin loading app jar
     */
    default void onStart() {
    }

    /**
     * Runs after basin starting
     */
    default void afterStart() {
    }

    /**
     * Runs before basin stopping
     */
    default void beforeStop() {
    }
}

File: src/main/java/net/liyze/basin/core/BasinBoot.kt
package net.liyze.basin.core

/**
 * Basin's Plugin/Boot SPI
 */
interface BasinBoot {
    /**
     * Runs on basin loading app jar
     */
    fun onStart() {}

    /**
     * Runs after basin starting
     */
    fun afterStart() {}

    /**
     * Runs before basin stopping
     */
    fun beforeStop() {}
}
